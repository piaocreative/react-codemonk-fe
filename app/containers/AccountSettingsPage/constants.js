export const NOTIFICATION_LIST = [
  'New jobs alert',
  'Interview requests',
  'Timesheet approvals',
  'Payment notifications',
  'New product features',
  'Marketing newsletters',
];

export const PREFERENCES = ['Email', 'SMS', 'Calls'];
