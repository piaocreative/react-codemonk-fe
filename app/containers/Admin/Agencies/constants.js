/*
 * AdminAgencies Constants
 */
import React from 'react';
import { TableSkeletonCol6 } from 'components/SkeletonLoader';

export const key = 'AdminAgencies';
export const CHANGE_AGENCY_STATUS = 'CodeMonk/AdminAgencies/CHANGE_AGENCY_STATUS';
export const DEFAULT_PAGE_NO = 1;
export const FILTER_KEYS = ['status'];

export const columns = [
  {
    name: 'SL No',
    selector: 'number',
    maxWidth: '110px',
    style: {
      paddingLeft: '40px',
    },
  },
  {
    name: 'Name',
    selector: 'name',
  },
  {
    name: 'Email',
    selector: 'email',
  },
  {
    name: 'Status',
    selector: 'status',
  },
  {
    name: 'Action',
    selector: 'action',
    allowOverflow: true,
    style: {
      paddingRight: '40px',
    },
  },
];

export const agencyActionStatusArray = [
  {
    label: 'All',
    value: -1,
  },
  {
    label: 'Active',
    value: 1,
  },
  {
    label: 'Suspend',
    value: 2,
  },
  {
    label: 'Unregistered',
    value: 0,
  },
];

export const LinearIndeterminate = () => (
  <div className="w-100 flex-column d-flex">
    <TableSkeletonCol6 cardCount={5} />
  </div>
);
