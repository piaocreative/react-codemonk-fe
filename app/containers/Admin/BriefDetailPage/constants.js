export const adminBriefTalentListingColumns = [
  {
    name: 'Name',
    selector: 'name',
    style: {
      paddingLeft: '40px',
    },
  },
  {
    name: 'Role',
    selector: 'role',
  },
  {
    name: 'Rate',
    selector: 'rate',
    style: {
      paddingRight: '40px',
    },
  },
];
