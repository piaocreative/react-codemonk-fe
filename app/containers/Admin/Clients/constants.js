/*
 * AdminClients Constants
 */

export const key = 'AdminClients';
export const CHANGE_CLIENT_STATUS = 'CodeMonk/AdminClients/CHANGE_STATUS';
export const DEFAULT_PAGE_NO = 1;
export const FILTER_KEYS = ['status'];

export const columns = [
  {
    name: 'SL No',
    selector: 'number',
    maxWidth: '110px',
    style: {
      paddingLeft: '40px',
    },
  },
  {
    name: 'Name',
    selector: 'name',
  },
  {
    name: 'Company Name',
    selector: 'compnayName',
  },
  {
    name: 'Email',
    selector: 'email',
  },
  {
    name: 'Status',
    selector: 'status',
  },
  {
    name: 'Action',
    selector: 'action',
    allowOverflow: true,
    style: {
      paddingRight: '40px',
    },
  },
];

export const clientActionStatusArray = [
  {
    label: 'All',
    value: -1,
  },
  {
    label: 'Active',
    value: 1,
  },
  {
    label: 'Suspend',
    value: 2,
  },
  {
    label: 'Unregistered',
    value: 0,
  },
];
