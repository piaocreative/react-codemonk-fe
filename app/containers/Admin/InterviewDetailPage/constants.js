export const key = 'AdminInterviewDetail';

export const CHANGE_TALENT_STATUS = 'CodeMonk/AdminInterviewDetail/CHANGE_TALENT_STATUS';

export const talentStatusArray = [
  {
    label: 'Hired',
    value: 1,
  },
  {
    label: 'Rejected',
    value: 0,
  },
];
