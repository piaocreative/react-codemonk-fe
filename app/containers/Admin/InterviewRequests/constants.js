/*
 * AdminInterviews Constants
 */

import React from 'react';
import { TableSkeletonCol6 } from 'components/SkeletonLoader';

export const key = 'AdminInterviews';

export const CHANGE_PROJECTS_ARRAY = 'CodeMonk/AdminInterviews/CHANGE_PROJECTS_ARRAY';
export const CHANGE_INTERVIEW_STATUS = 'CodeMonk/AdminInterviews/CHANGE_INTERVIEW_STATUS';

export const DEFAULT_PAGE_NO = 1;
export const FILTER_KEYS = ['status'];

export const projectStatusArray = [
  {
    label: 'All',
    value: -1,
  },
  {
    label: 'Requested',
    value: 0,
  },
  {
    label: 'In Progress',
    value: 1,
  },
  {
    label: 'Done',
    value: 2,
  },
];

export const columns = [
  {
    name: 'SL No',
    selector: 'number',
    maxWidth: '110px',
    style: {
      paddingLeft: '40px',
    },
  },
  {
    name: 'Date Requested',
    selector: 'dateRequested',
  },
  {
    name: 'Client Name',
    selector: 'clientName',
  },
  {
    name: 'Company Name',
    selector: 'companyName',
  },
  {
    name: 'Project Name',
    selector: 'name',
  },
  {
    name: 'Talent Name',
    selector: 'talentName',
  },
  {
    name: 'Status',
    selector: 'status',
    minWidth: '150px',
    allowOverflow: true,
  },
  {
    name: '',
    selector: 'action',
    maxWidth: '100px',
    style: {
      paddingRight: '40px',
    },
  },
];

export const LinearIndeterminate = () => (
  <div className="w-100 flex-column d-flex">
    <TableSkeletonCol6 cardCount={5} />
  </div>
);
