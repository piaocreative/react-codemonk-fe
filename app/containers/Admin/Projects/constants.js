/*
 * AdminProjects Constants
 */

export const key = 'AdminProjects';
export const FILTER_KEYS = ['status'];

export const DEFAULT_PAGE_NO = 1;
export const CHANGE_PROJECTS_ARRAY = 'CodeMonk/AdminProjects/CHANGE_PROJECTS_ARRAY';
export const CHANGE_PROJECT_STATUS = 'CodeMonk/AdminProjects/CHANGE_STATUS';
export const ADD_PROJECT = 'CodeMonk/AdminProjects/ADD_PROJECT';

export const projectStatusPopup = [
  {
    label: 'Proposed',
    value: '1',
  },
  {
    label: 'Discovery',
    value: '2',
  },
  {
    label: 'Kick-off',
    value: '3',
  },
  {
    label: 'In Progress',
    value: '4',
  },
  {
    label: 'On Hold',
    value: '5',
  },
  {
    label: 'Suspended',
    value: '6',
  },
  {
    label: 'Closed',
    value: '7',
  },
];

const adminProjectsColumns1 = [
  {
    name: 'SL No',
    selector: 'number',
    maxWidth: '110px',
    style: {
      paddingLeft: '40px',
    },
  },
  {
    name: 'Project Name',
    selector: 'name',
  },
  {
    name: 'Client Name',
    selector: 'clientName',
  },
  {
    name: 'Company Name',
    selector: 'companyName',
  },
];
const adminProjectsColumns2 = [
  {
    name: 'Start Date',
    selector: 'startDate',
  },
  {
    name: 'End Date',
    selector: 'endDate',
  },
  {
    name: 'Status',
    selector: 'status',
    maxWidth: '165px',
    allowOverflow: true,
  },
  {
    name: '',
    selector: 'action',
    style: {
      paddingRight: '40px',
      justifyContent: 'center',
    },
  },
];
export const adminProjectsColumns = [...adminProjectsColumns1, ...adminProjectsColumns2];
