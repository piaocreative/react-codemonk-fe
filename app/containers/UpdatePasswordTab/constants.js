export const key = 'updatePasswordForm';

const prefix = 'CHANGE_PASSWORD_FORM';

export const CHANGE_OLD_PASSWORD = `${prefix}CHANGE_OLD_PASSWORD`;
export const CHANGE_PASSWORD = `${prefix}CHANGE_PASSWORD`;
export const CHANGE_CONFIRM_PASSWORD = `${prefix}CHANGE_CONFIRM_PASSWORD`;
export const CHANGE_TALENT_PASSWORD = `${prefix}CHANGE_TALENT_PASSWORD`;
