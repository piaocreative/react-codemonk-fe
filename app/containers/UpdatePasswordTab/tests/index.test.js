import React from 'react';
import { shallow } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';
import configureStore from 'redux-mock-store';
import request from 'utils/request';
import { LOAD_REPOS } from 'containers/App/constants';
import { UpdatePasswordTab as MainForm, mapDispatchToProp } from '../index';
import initialState from '../reducer';
import { CHANGE_OLD_PASSWORD, CHANGE_PASSWORD, CHANGE_CONFIRM_PASSWORD, CHANGE_TALENT_PASSWORD } from '../constants';
jest.mock('utils/request');
let store;
const mockStore = configureStore();
const props = {
  oldPassword: '',
  password: '',
  confirmPassword: '',

  dispatch: jest.fn(),
  handleSubmit: jest.fn(),

  invalid: false,
  loading: false,
  responseSuccess: false,
  responseError: false,
  history: { push: jest.fn(), replace: jest.fn() },
  location: {},

  onChangeOldPassword: jest.fn(),
  onChangePassword: jest.fn(),
  onChangeConfirmPassword: jest.fn(),
  onSubmitTalentPassword: jest.fn(),
};
const getWrapper = () => shallow(<MainForm store={store} {...props} />);
const getInstance = () => getWrapper().instance();
const intializeSetup = () => {
  store = mockStore(initialState);
  request.mockImplementation(() => Promise.resolve({ status: 1 }));
};

describe('<MainForm />', () => {
  const renderer = new ShallowRenderer();
  it('should render and match the snapshot', () => {
    renderer.render(<MainForm store={store} {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});

describe('test functions', () => {
  intializeSetup();
  getWrapper();

  test('Testing if onClickIcon with passwordType', () => {
    const onClickIcon = jest.spyOn(getInstance(), 'onClickIcon');
    onClickIcon('passwordType');
    expect(onClickIcon).toHaveBeenCalledTimes(1);
  });
  test('Testing if onClickIcon confirmPasswordType', () => {
    const onClickIcon = jest.spyOn(getInstance(), 'onClickIcon');
    onClickIcon('confirmPasswordType');
    expect(onClickIcon).toHaveBeenCalledTimes(1);
  });

  test('Testing if handleSubmitButton ', () => {
    const handleSubmitButton = jest.spyOn(getInstance(), 'handleSubmitButton');
    handleSubmitButton(false, true);
    expect(handleSubmitButton).toHaveBeenCalledTimes(1);
  });

  test('Testing if handleSubmitButton with else', () => {
    const handleSubmitButton = jest.spyOn(getInstance(), 'handleSubmitButton');
    handleSubmitButton(true, true);
    expect(handleSubmitButton).toHaveBeenCalledTimes(1);
  });
});

describe('mapDispatchToProp Testing', () => {
  intializeSetup();
  getWrapper();
  test('Testing if  CHANGE_OLD_PASSWORD are dispatched correctly', () => {
    const dispatch = jest.fn();
    mapDispatchToProp(dispatch).onChangeOldPassword({ target: { value: 'password' } });
    expect(dispatch.mock.calls[0][0]).toEqual({
      payload: 'password',
      type: CHANGE_OLD_PASSWORD,
    });
  });

  test('Testing if  CHANGE_PASSWORD are dispatched correctly', () => {
    const dispatch = jest.fn();
    mapDispatchToProp(dispatch).onChangePassword({ target: { value: 'password' } });
    expect(dispatch.mock.calls[0][0]).toEqual({
      payload: 'password',
      type: CHANGE_PASSWORD,
    });
  });

  test('Testing if  CHANGE_CONFIRM_PASSWORD are dispatched correctly', () => {
    const dispatch = jest.fn();
    mapDispatchToProp(dispatch).onChangeConfirmPassword({ target: { value: 'password' } });
    expect(dispatch.mock.calls[0][0]).toEqual({
      payload: 'password',
      type: CHANGE_CONFIRM_PASSWORD,
    });
  });

  test('Testing if the  LOAD_REPOS events are dispatched correctly', () => {
    const dispatch = jest.fn();
    const event = {
      preventDefault: jest.fn(),
    };
    mapDispatchToProp(dispatch).onSubmitTalentPassword(event);
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: LOAD_REPOS,
    });
  });

  test('Testing if the  CHANGE_TALENT_PASSWORD events are dispatched correctly', () => {
    const dispatch = jest.fn();
    const event = {
      preventDefault: jest.fn(),
    };
    mapDispatchToProp(dispatch).onSubmitTalentPassword(event);
    expect(dispatch.mock.calls[1][0]).toEqual({
      type: CHANGE_TALENT_PASSWORD,
    });
  });
});
