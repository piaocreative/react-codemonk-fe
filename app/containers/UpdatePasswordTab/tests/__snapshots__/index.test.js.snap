// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainForm /> should render and match the snapshot 1`] = `
<React.Fragment>
  <ForwardRef(H4)
    className="newH4 mt-4 mb-3"
    opacityVal="0.5"
  >
    <FormattedMessage
      defaultMessage="Change your password"
      id="CodeMonk.containers.containerMessage.changePassword"
      values={Object {}}
    />
  </ForwardRef(H4)>
  <form>
    <Row
      tag="div"
    >
      <Col
        className="col-md-6"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <FormGroup
          tag="div"
        >
          <ForwardRef(Label__FormLabel)>
            <FormattedMessage
              defaultMessage="Current Password"
              id="CodeMonk.containers.containerMessage.labelOldPassword"
              values={Object {}}
            />
          </ForwardRef(Label__FormLabel)>
          <ForwardRef
            component={[Function]}
            disabled={false}
            name="oldPassword"
            normalize={[Function]}
            onChange={[MockFunction]}
            onClick={[Function]}
            placeholder="Enter current password"
            type="password"
            validate={
              Array [
                [Function],
              ]
            }
            value=""
          />
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <ForwardRef(Label__FormLabel)>
            <FormattedMessage
              defaultMessage="New Password"
              id="CodeMonk.components.ResetPasswordPage.labelNewPassword"
              values={Object {}}
            />
          </ForwardRef(Label__FormLabel)>
          <ForwardRef(signup-styles__PasswordFieldWrapper)>
            <div
              className="w-100"
            >
              <ForwardRef
                component={[Function]}
                disabled={false}
                errorMessageToShow={true}
                id="tooltipPassword"
                name="password"
                normalize={[Function]}
                onBlur={[Function]}
                onChange={[MockFunction]}
                onClick={[Function]}
                onFocus={[Function]}
                placeholder="Enter new password"
                type="password"
                validate={
                  Array [
                    [Function],
                    [Function],
                  ]
                }
                value=""
              />
            </div>
            <div
              className="tooltip-wrapper"
            >
              <span
                id="tooltipPassword"
              />
              <Tooltip
                autohide={true}
                className="d-none d-md-block"
                delay={
                  Object {
                    "hide": 250,
                    "show": 0,
                  }
                }
                hideArrow={false}
                isOpen={false}
                placement="right"
                placementPrefix="bs-tooltip"
                target="tooltipPassword"
                toggle={[Function]}
              >
                <ForwardRef(signup-styles__ToolTipUlWrapper)>
                  <ul>
                    <li
                      className=""
                    >
                      <span>
                        <FormattedMessage
                          defaultMessage="Lower case letters (a-z)"
                          id="CodeMonk.containers.containerMessage.lowerCase"
                          values={Object {}}
                        />
                      </span>
                    </li>
                    <li
                      className=""
                    >
                      <span>
                        <FormattedMessage
                          defaultMessage="Upper case letters (A-Z)"
                          id="CodeMonk.containers.containerMessage.upperCase"
                          values={Object {}}
                        />
                      </span>
                    </li>
                    <li
                      className=""
                    >
                      <span>
                        <FormattedMessage
                          defaultMessage="Numbers (i.e 0-9)"
                          id="CodeMonk.containers.containerMessage.numbersInString"
                          values={Object {}}
                        />
                      </span>
                    </li>
                    <li
                      className=""
                    >
                      <span>
                        <FormattedMessage
                          defaultMessage="Special Characters (i.e ^Â£$%&*@)"
                          id="CodeMonk.containers.containerMessage.specialCharacter"
                          values={Object {}}
                        />
                      </span>
                    </li>
                    <li
                      className=""
                    >
                      <span>
                        <FormattedMessage
                          defaultMessage="Min 8 Characters long"
                          id="CodeMonk.containers.containerMessage.stringLength"
                          values={Object {}}
                        />
                      </span>
                    </li>
                  </ul>
                </ForwardRef(signup-styles__ToolTipUlWrapper)>
              </Tooltip>
            </div>
          </ForwardRef(signup-styles__PasswordFieldWrapper)>
        </FormGroup>
        <FormGroup
          tag="div"
        >
          <ForwardRef(Label__FormLabel)>
            <FormattedMessage
              defaultMessage="Confirm Password"
              id="CodeMonk.components.ResetPasswordPage.labelConfirmPassword"
              values={Object {}}
            />
          </ForwardRef(Label__FormLabel)>
          <ForwardRef
            component={[Function]}
            disabled={false}
            name="confirmPassword"
            normalize={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            placeholder="Repeat new password"
            type="password"
            validate={
              Array [
                [Function],
                [Function],
              ]
            }
            value=""
          />
        </FormGroup>
      </Col>
    </Row>
    <Button
      className="mt-3 btn-primary btn-submit"
      disabled={false}
    >
      <FormattedMessage
        defaultMessage="Change password"
        id="CodeMonk.components.ResetPasswordPage.changePassword"
        values={Object {}}
      />
    </Button>
  </form>
</React.Fragment>
`;
